sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/pipeline/TelemetryItemSerialization.java

```diff
@@ -3,53 +3,91 @@
 
 package com.azure.monitor.opentelemetry.exporter.implementation.pipeline;
 
-import com.azure.core.util.logging.ClientLogger;
-import com.azure.core.util.logging.LogLevel;
+import com.azure.json.JsonProviders;
+import com.azure.json.JsonReader;
+import com.azure.json.JsonToken;
+import com.azure.json.JsonWriter;
 import com.azure.monitor.opentelemetry.exporter.implementation.models.TelemetryItem;
-import com.fasterxml.jackson.annotation.JsonInclude;
-import com.fasterxml.jackson.core.JsonGenerator;
-import com.fasterxml.jackson.core.io.SerializedString;
-import com.fasterxml.jackson.databind.MappingIterator;
-import com.fasterxml.jackson.databind.ObjectMapper;
-import com.fasterxml.jackson.databind.SerializationFeature;
 
 import java.io.ByteArrayInputStream;
 import java.io.ByteArrayOutputStream;
 import java.io.IOException;
-import java.io.StringWriter;
 import java.nio.ByteBuffer;
+import java.util.ArrayList;
 import java.util.List;
 import java.util.zip.GZIPInputStream;
 import java.util.zip.GZIPOutputStream;
 
 public final class TelemetryItemSerialization {
 
-    private static final ObjectMapper mapper = createObjectMapper();
     private static final AppInsightsByteBufferPool byteBufferPool = new AppInsightsByteBufferPool();
-    private static final ClientLogger logger = new ClientLogger(TelemetryItemSerialization.class);
 
-    public static List<TelemetryItem> deserialize(byte[] data) {
-        return deserializeAlreadyDecoded(decode(data));
+    // serialize an array of TelemetryItems to an array of byte buffers
+    public static List<ByteBuffer> serialize(List<TelemetryItem> telemetryItems) {
+        try {
+            ByteBufferOutputStream out = writeTelemetryItemsAsByteBufferOutputStream(telemetryItems);
+            out.close(); // closing ByteBufferOutputStream is a no-op, but this line makes LGTM happy
+            List<ByteBuffer> byteBuffers = out.getByteBuffers();
+            for (ByteBuffer byteBuffer : byteBuffers) {
+                byteBuffer.flip();
+            }
+            return out.getByteBuffers();
+        } catch (IOException e) {
+            throw new IllegalStateException("Failed to serialize list of TelemetryItems to List<ByteBuffer>", e);
+        }
     }
 
     // visible for testing
-    // deserialize raw bytes to a list of TelemetryItem without decoding
-    public static List<TelemetryItem> deserializeAlreadyDecoded(byte[] data) {
-        try {
-            MappingIterator<TelemetryItem> iterator = mapper.readerFor(TelemetryItem.class).readValues(data);
-            return iterator.readAll();
-        } catch (Exception e) {
-            throw new IllegalStateException("Failed to deserialize byte[] to a list of TelemetryItems", e);
+    // deserialize multiple TelemetryItem raw bytes with newline delimiters to a list of TelemetryItems
+    public static List<TelemetryItem> deserialize(byte[] rawBytes) {
```

Comment:
can we move this method to a test class since looks like only called by tests?

Reply:
i prefer this to be placed right next to serialize.